### PR TITLE
[ASGenerator] Fixed 'Assign statement to var' for AS3: 'delete' and 'typeof', Haxe: 'cast' and 'untyped'. fixes #1908. fixes #1920.

### LINKED ISSUE #1908: [Bug][AS3][ASGenerator] Wrong result after `Assign statement to var` for `typeof 1`
```actionscript
typeof 1; |
```
`|` - curstor
expected result: `var string:String = typeof 1`
actual result: `typeof var number:Number = 1;`

Depends on #1906 #1766

Note:
AS3 string operators: 
+ lvalue as rvalue -> T
+ delete rvalue -> Boolean
+ lvalue in rvalue -> Boolean
+ lvalue is rvalue -> Boolean
+ lvalue instanceof rvalue -> Boolean
+ new rvalue -> T
+ typeof rvalue -> String

Haxe string operators:
+ new rvalue-> T
+ cast rvalue-> Dynamic
+ untyped rvalue -> Dynamic

### LINKED ISSUE #1920: [Bug][Haxe][ASGenerator] Wrong result after `Assign statement to var` for `cast v`
```haxe
var s:Dynamic = "";
cast s;|
```
`|` - cursor
expected result: `var dynamicValue:Dynamic = cast s;`
actual result: `cast var s1:Dynamic = s;`

### PR DESCRIPTION
**Pull request by SlavaRa**

### CODE REVIEW

**Comment by Neverbirth** [comment]
@elsassph maybe you can check?

File: External/Plugins/ASCompletion/Completion/ASComplete.cs [hunk 1]
```diff
@@ -2627,8 +2627,13 @@ private static ASResult EvalExpression(string expression, ASExpr context, FileMo
         {
             ASResult notFound = new ASResult {Context = context};
             if (string.IsNullOrEmpty(expression)) return notFound;
-
             var ctx = ASContext.Context;
+            ClassModel type;
+            if (!string.IsNullOrEmpty(context.WordBefore))
```
Comment (elsassph) [inline External/Plugins/ASCompletion/Completion/ASComplete.cs]: This kind of cases should have a little comment like `// case: delete a / typeof a

**Comment by Meychi** [comment]
Conficts. Will be merged when fixed.

File: External/Plugins/ASCompletion/Completion/ASComplete.cs (hunk 1 above)
Comment (SlavaRa) [reply External/Plugins/ASCompletion/Completion/ASComplete.cs]: and `cast`, and `untyped` and *OtherOperators*... )

Comment (SlavaRa) [reply External/Plugins/ASCompletion/Completion/ASComplete.cs]: fixed

**Pull request merged**